libc/stdio_printf.c

```diff
@@ -0,0 +1,3336 @@
+	}
+}
+
+
+TEST(stdio_printf_rest, zero_float)
+{
+	int i;
+	const char *format = "%0.2f %04.2f %08.2f";
+	const float values[] = { 0, -1, 1, -64.321, 64.321, 98765.43120 };
+	char *expect[] = { "0.00 0.00 00000.00", "-1.00 -1.00 -0001.00", "1.00 1.00 00001.00", "-64.32 -64.32 -0064.32", "64.32 64.32 00064.32", "98765.43 98765.43 98765.43" };
+
+	for (i = 0; i < (sizeof(values) / sizeof(values[0])); i++) {
+
+		dtest(format, values[i] COMMA values[i] COMMA values[i], expect[i]);
+		ftest(format, values[i] COMMA values[i] COMMA values[i], expect[i]);
+		stdtest(format, values[i] COMMA values[i] COMMA values[i], expect[i]);
+		sntest(format, values[i] COMMA values[i] COMMA values[i], expect[i]);
+		stest(format, values[i] COMMA values[i] COMMA values[i], expect[i]);
+
+		vdtest(expect[i], format, values[i], values[i], values[i]);
+		vftest(expect[i], format, values[i], values[i], values[i]);
+		vtest(expect[i], format, values[i], values[i], values[i]);
+		vsntest(expect[i], format, values[i], values[i], values[i]);
+		vstest(expect[i], format, values[i], values[i], values[i]);
+	}
+}
+
+
+TEST(stdio_printf_rest, numbered_argument)
+{
+/* Disabled because of #719 issue: https://github.com/phoenix-rtos/phoenix-rtos-project/issues/719 */
+#ifdef __phoenix__
+	TEST_IGNORE_MESSAGE("#719 issue");
+#endif
+	const char *format = "%3$d %2$d %1$d";
+
+	dtest(format, 1 COMMA 2 COMMA 3, "3 2 1");
+	ftest(format, 1 COMMA 2 COMMA 3, "3 2 1");
+	stdtest(format, 1 COMMA 2 COMMA 3, "3 2 1");
+	sntest(format, 1 COMMA 2 COMMA 3, "3 2 1");
+	stest(format, 1 COMMA 2 COMMA 3, "3 2 1");
+
+	vdtest("3 2 1", format, 1 COMMA 2 COMMA 3);
+	vftest("3 2 1", format, 1 COMMA 2 COMMA 3);
+	vtest("3 2 1", format, 1 COMMA 2 COMMA 3);
+	vsntest("3 2 1", format, 1 COMMA 2 COMMA 3);
+	vstest("3 2 1", format, 1 COMMA 2 COMMA 3);
+}
+
+
+TEST(stdio_printf_rest, snprintf_truncation)
+{
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wformat-truncation"
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 0, "%d", 1234567890);
+	TEST_ASSERT_EQUAL_STRING("", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 6, "%d", 1234567890);
+	TEST_ASSERT_EQUAL_STRING("12345", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 6, "%f", 1.23456789);
+	TEST_ASSERT_EQUAL_STRING("1.234", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 0, "%s", "abcdefighjklmnop");
+	TEST_ASSERT_EQUAL_STRING("", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 6, "%s", "abcdefighjklmnop");
+	TEST_ASSERT_EQUAL_STRING("abcde", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 3, "%6s", "abc");
+	TEST_ASSERT_EQUAL_STRING("  ", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 6, "%6s", "abc");
+	TEST_ASSERT_EQUAL_STRING("   ab", buff);
+
+	memset(buff, 0, sizeof(buff));
+	snprintf(buff, 7, "%6s", "abc");
+	TEST_ASSERT_EQUAL_STRING("   abc", buff);
+
+#pragma GCC diagnostic pop
+}
+
+
+TEST(stdio_printf_rest, errnos)
```

Comment:
Please write some short comments why the rest of errnos are not checked - for example: `/* EILSEQ not tested, because locales are not yet supported on Phoenix-RTOS`


I think that catching `EOVERFLOW` could be possible.